File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        l = 0
        r = len(cardPoints) - k - 1
        k_sum = sum(cardPoints[l:r+1])
        total_sum = k_sum + sum(cardPoints[r+1:])
        curr_max = float('-inf')
        while(r < len(cardPoints)-1):
            print(l,r,k_sum,curr_max)
            curr_max = max(curr_max, total_sum - k_sum)
            r+=1
            k_sum+=cardPoints[r]
            k_sum-=cardPoints[l]
            l+=1
        return max(curr_max, total_sum - k_sum)

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int curr = 0;
        int N = cardPoints.length;
        for(int i = 0;i<N-k;i++){
            curr+=cardPoints[i];
        }
        int l = 0;
        int r = N-k-1;
        int mini = curr;
        while(r < N-1){
            if(curr < mini){
                mini = curr;
            }
            r+=1;
            curr+=cardPoints[r];
            curr-=cardPoints[l];
            l+=1;
        }
        if(curr < mini){
            mini = curr;
        }
        int total = 0;
        for(int i = 0;i<N;i++){
            total+=cardPoints[i];
        }
        return total - mini;
    }
}